### PR TITLE
fix(ingest): profiling - Fixing issue with the wrong timestamp stored in check

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def generate_wu_from_profile_requests(
             # We don't add to the profiler state if we only do table level profiling as it always happens
             if self.state_handler and not request.profile_table_level_only:
                 self.state_handler.add_to_state(
-                    dataset_urn, int(datetime.utcnow().timestamp() * 1000)
+                    dataset_urn, int(datetime.now().timestamp() * 1000)
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py:234]: Why is this change needed ? I think, we should always persist UTC timestamps. In fact, if last_altered from database query is non-utc, we should convert it to UTC on database side OR locally in python code before comparing with timestamp from profiling state. WDYT ?

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py [hunk 2]
```diff
@@ -328,9 +328,9 @@ def get_tables_for_dataset(
                 name=table.table_name,
                 created=table.created,
                 last_altered=datetime.fromtimestamp(
-                    table.last_altered / 1000, tz=timezone.utc
+                    table.get("last_altered") / 1000, tz=timezone.utc
                 )
-                if "last_altered" in table
+                if "last_altered" in table.keys() and table.get("last_altered")
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py:333]:
```suggestion
                if table.get("last_altered") is not None
```

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py:234]: in general, we should use timezone-aware datetimes

```py
datetime.now(timezone.utc)
```

however, because we're calling .timestamp() immediately, these two are basically the same - probably ok for now

```
>>> datetime.now().timestamp()
1673267826.735305
>>> datetime.now(timezone.utc).timestamp()
1673267828.30003
>>>
```

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py:234]: Ohh, got it.
